core/src/main/java/org/apache/spark/shuffle/sort/BypassMergeSortShuffleWriter.java

```diff
@@ -102,6 +104,7 @@
       IndexShuffleBlockResolver shuffleBlockResolver,
       BypassMergeSortShuffleHandle<K, V> handle,
       int mapId,
+      TaskContext taskContext,
```

Comment:
If we only need `TaskContext` to get `shuffleGenerationId`, I think it's better to get `shuffleGenerationId` at caller side.

Reply:
ah i see, you wanna be consistent with `UnsafeShuffleWriter`. Then it's fine.

Reply:
Yeah.